src/simplay-web/src/Entity.ts

```diff
@@ -37,27 +66,92 @@ export async function createEntities(context: SimplayContext) {
     if (frames.length === 0) {
       throw new Error(`No frames found for visual ${entity.visual}`);
     }
-    const loadedFrames: PIXI.Texture[] = [];
-    for (const frame of frames) {
-      const loadedFrame = await PIXI.Texture.fromURL(frame);
-      loadedFrames.push(loadedFrame);
-    }
-    const sprite = new PIXI.AnimatedSprite(loadedFrames);
-    sprite.animationSpeed = 0;
-    sprite.loop = false;
-    const { width, height } = sprite.getBounds();
-    const scale = Math.min(
-      context.tileWidth / width,
-      context.tileHeight / height
-    );
-    // multiplying by 0.5 is a design choice to make the entities look a bit smaller
-    sprite.scale.set(scale * 0.5, scale * 0.5);
-    sprite.name = entity.id;
-    if (entity.tint && entity.tint !== 0xffffff) {
-      sprite.tint = entity.tint;
+    const sprite = await createAnimatedSprite(context, entity, frames);
+    const text = createDecoratingText(entity);
+    text.y = sprite.y + sprite.height + 1;
+    text.x = sprite.x + sprite.width / 2;
+
+    const container = new PIXI.Container();
+    container.name = entity.id;
+    container.addChild(sprite);
+    container.addChild(text);
+    container.visible = false;
+
+    let displayEntity = {
+      animatedSprite: sprite,
+      decoratingText: text,
+      container: container,
+    } as DisplayEntity;
+
+    if (
+      entity.type === 'CONTAINER' ||
+      entity.type === 'STORE' ||
+      entity.type === 'RESOURCE'
+    ) {
+      const informationText = createInformationText(entity);
+      informationText.y = sprite.y + sprite.height + 6;
+      informationText.x = sprite.x + sprite.width / 2;
+      container.addChild(informationText);
+      displayEntity = {
+        ...displayEntity,
+        informationText: informationText,
+      } as ExtendedDisplayEntity;
     }
-    sprite.visible = false;
 
-    context.entityContainer.addChild(sprite);
+    context.entityContainer.addChild(container);
+    context.entityDictionary[entity.id] = displayEntity;
+  }
+}
+
+function createDecoratingText(entity: Entity): PIXI.Text {
+  const text = new PIXI.Text(entity.id, {
+    fontFamily: 'Arial',
+    fontSize: 12,
+    fill: 0xffffff,
+    align: 'center',
+  });
+  text.anchor.set(0.5, 0.5);
+  text.name = `${entity.id}-text`;
+  return text;
+}
+
+function createInformationText(entity: Entity): PIXI.Text {
+  const text = new PIXI.Text(entity.id, {
+    fontFamily: 'Arial',
+    fontSize: 12,
+    fill: 0xffffff,
+    align: 'center',
+  });
+  text.anchor.set(0.5, 0.5);
```

Comment:
To me it makes sense - but I'm the one who's written the code. If it's clear to you, I think we can leave it undocumented, otherwise lets document it. I'm not really a big fan of comments in code.

Reply:
leave it undocumented, just checked the pixi docs: https://pixijs.download/dev/docs/PIXI.Text.html#anchor reading this its clear what 0.5, 0.5 does. But we could extract them into a constant.